build.gradle

```diff
@@ -213,9 +213,9 @@ tasks.register("verifyVersions") {
  * after the backport of the backcompat code is complete.
  */
 
-boolean bwc_tests_enabled = true
+boolean bwc_tests_enabled = false
 // place a PR link here when committing bwc changes:
-String bwc_tests_disabled_issue = ""
+String bwc_tests_disabled_issue = "https://github.com/elastic/elasticsearch/pull/72933"
```

Comment:
Which BWC tests need to be disabled?

Reply:
I think it was the REST yaml test for master -> 7.x that blew up because the search_after param is new.
Historically I've tried merging multiple PRs with temporary "if version = " checks while we do the backport dance and then clean them up but nearly always end up breaking builds so I opted for this big switch.

Reply:
I'm going to ping @pgomulka on this to get the best way of handling forward YAML tests - I think we can add the specific failing tests to a blocklist somewhere to avoid the big red button.

Reply:
My current plan was:
1) Push this commit to master with all BWC turned off
2) Backport to 7.x but with all BWC flag turned back on
3) Push commit to master with BWC flag turned back on.


These are the less sweeping changes I _think_ I'd need. This is to modify the stream protocol for a 7.14 (ie currently unreleased) feature.
1. Make a change targeting master with 
  a) added BWC in/out stream version checks for on or after the current version (8.0) in master
  b) added skip to the yaml REST test for " - 7.99.99". 
3. Integrate the change into master.
4. Backport the change to 7.x, minus the stream version checks and rest skip header (instead keeping the existing REST tests skip in 7.x which already has  " - 7.13.99").
5. Integrate this change into 7.x.
6. Push one more change to master that removes the in/out BWC version checks and removes the rest skips  - (we only test BWC for 8.0 and last 7.x?).

@pgomulka can you verify?

Reply:
I hadn't spotted that this was a change to the stream API, I'd thought it was just the REST compat tests.  In which case, +1 to your original plan.

Reply:
I am probably not expert enough to recommend the best approach here. 
but I wonder if simply adding 
```
tasks.named("bwcTest").configure {

    systemProperty 'tests.rest.blacklist', [
    ]
}
```
would not work?
Could @elastic/es-delivery advice us here?

Reply:
@markharwood I think your plan is good and it should not affect the CI for others

Reply:
> I think your plan is good and it should not affect the CI for others

Thanks for the feedback @pgomulka. Did you mean plan 1 (switching the global BWC flag)or plan 2 with adding the temporary version checking code (documented [above](https://github.com/elastic/elasticsearch/pull/72933#discussion_r632393931))?

Reply:
In my second plan I don't think I need to add the version checks in the streaming code if we disable the rest tests (step 1b).
I've just tried pushing a change based on this approach to see if it builds OK.